buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -355,21 +355,29 @@ private static int lastIndexOf(ByteBuf buffer, int fromIndex, int toIndex, byte
 
     /**
      * Encode a {@link CharSequence} in <a href="http://en.wikipedia.org/wiki/UTF-8">UTF-8</a> and write
+     * it to a {@link ByteBuf} allocated with {@code alloc}.
+     * @param alloc The allocator used to allocate a new {@link ByteBuf}.
+     * @param seq The characters to write into a buffer.
+     * @return The {@link ByteBuf} which contains the <a href="http://en.wikipedia.org/wiki/UTF-8">UTF-8</a> encoded
+     * result.
+     */
+    public static ByteBuf writeUtf8(ByteBufAllocator alloc, CharSequence seq) {
+        // UTF-8 uses max. 3 bytes per char, so calculate the worst case.
+        ByteBuf buf = alloc.buffer(seq.length() * 3);
```

Comment:
> i think we need to ensure releasing the buffer on an error

writeUtf8 no longer throws to be consistent with `String.getBytes()`.

> Also, is the constant not defined somewhere already?

I don't believe so. I can define one locally in this file.